server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java

```diff
@@ -0,0 +1,154 @@
+ **/
+public class ForkedInfinispanServerDriver extends AbstractInfinispanServerDriver {
+    private static final Log log = org.infinispan.commons.logging.LogFactory.getLog(ForkedInfinispanServerDriver.class);
+    private List<ForkedServer> forkedServers = new ArrayList<>();
+
+
+    protected ForkedInfinispanServerDriver(InfinispanServerTestConfiguration configuration) {
+        super(configuration, InetAddress.getLoopbackAddress());
+    }
+
+    @Override
+    public void prepare(String name) {
+        //do nothing
+    }
+
+    @Override
+    public void start(String name) {
+        log.infof("Starting server %s", name);
+        start(name, null, configuration.configurationFile());
+        log.infof("Started server %s", name);
+    }
+
+    @Override
+    protected void start(String name, File rootDir, String configurationFile) {
+        String[] serverLocations = TestSystemPropertyNames.INFINISPAN_SERVER_HOME.replaceAll("\\s+", "").split(",");
+        TestSystemPropertyNames.INFINISPAN_SERVER_CONFIGURATION = TestSystemPropertyNames.INFINISPAN_SERVER_CONFIGURATION == null ? configurationFile : TestSystemPropertyNames.INFINISPAN_SERVER_CONFIGURATION;
+        if(serverLocations.length != configuration.numServers()) {
+            throw new IllegalArgumentException("configuration.numServers should be the same " +
+                    "of the number of servers declared on org.infinispan.test.server");
+        }
+        for (int i = 0; i < configuration.numServers(); i++) {
+            ForkedServer server = new ForkedServer(serverLocations[i])
+                    .setServerConfiguration(TestSystemPropertyNames.INFINISPAN_SERVER_CONFIGURATION)
+                    .setPortsOffset(i)
+                    .start();
+            forkedServers.add(server);
+            copyArtifactsToUserLibDir(server.getServerLib());
+            log.info(forkedServers.get(0).printServerLog());
+        }
+    }
+
+    @Override
+    /**
+     * Stop all cluster
+     */
+    protected void stop() {
+        sync(getRestClient(0).cluster().stop());
+    }
+
+    @Override
+    /**
+     * Stop a specific server
+     */
+    public void stop(int server) {
+        sync(getRestClient(0).server().stop());
+    }
+
+    @Override
+    public boolean isRunning(int server) {
+        try {
+            sync(getRestClient(server).server().configuration());
+        } catch (RuntimeException r) {
+            return !(Util.getRootCause(r) instanceof ConnectException);
+        }
+        return true ;
+    }
+
+    @Override
+    public InetSocketAddress getServerSocket(int server, int port) {
+        return new InetSocketAddress(getServerAddress(server), getServerPort(server));
+    }
+
+    @Override
+    public InetAddress getServerAddress(int server) {
+        return Exceptions.unchecked(() -> InetAddress.getByName("localhost"));
+    }
+
+    @Override
+    public void pause(int server) {
+        throw new UnsupportedOperationException();
```

Comment:
Why not `kill -STOP`

Reply:
@tristantarrant IMO with a forked process makes more sense to totally stop the process, release the port opened and if needed create a new process.

Reply:
IMO, the forked mode shouldn't support pause or continue.
`EmbeddedInfinispanServerDriver` doesn't support `pause` or `continue`.
If we need to deal with those scenarios we must use `ContainerInfinispanServerDriver`
The `throw new UnsupportedOperationException();` looks good to me.